[resampler] Use ffmpeg for quiet noise generation

popcornmix opened the pull request:
Opening the GPU resampler is relatively expensive compared to the small amount of work done here.
This function takes ~110ms when using GPU resampler and about 14ms with ffmpeg resample on Pi 2.

Provide a way of flagging that GPU acceleration is not required

popcornmix committed:
[resampler] Use ffmpeg for quiet noise generation

Opening the GPU resampler is relatively expensive compared to the small amount of work done here.
This function takes ~110ms when using GPU resampler and about 14ms with ffmpeg resample on Pi 2.

Provide a way of flagging that GPU acceleration is not required

popcornmix commented:
ping @FernetMenta

fritsch commented:
Not @FernetMenta but looks fine.

popcornmix commented:
@fritsch comments are welcome too!
jenkins build this please

FernetMenta commented on xbmc/cores/AudioEngine/AEResampleFactory.h:
```diff
@@ -26,10 +26,18 @@ class IAEResample;
+ * Bit options to pass to CAEResampleFactory::Create
+ */
+enum AEStreamOptions {
+  /* This is a quick resample job (e.g. resample a single noise packet) and may not be worth using GPU acceleration */
```
coding style

popcornmix replied:
It matches:
https://github.com/xbmc/xbmc/blob/master/xbmc/cores/AudioEngine/Interfaces/AEStream.h#L29
I can change if desired.

FernetMenta replied:
please change the other location too. apart from this the pr is ok

popcornmix commented:
Updated with comments. I've also added the "quick" flag to GUI sound resampling.
On Pi2 I measured about 105ms with GPU acceleration and about 45ms with ffmpeg.

popcornmix commented:
@FernetMenta @fritsch 
Okay to merge?

fritsch commented:
Yes of course. Thx much!
